### PR TITLE
chore: Bump cryptography

### PR DESCRIPTION
**Pull request by suryadev99**
Fixing security vulnerability: updating cryptography package to fix vulnerability

### SUMMARY
 Updating cryptography package to 41.0.0

image

### TESTING INSTRUCTIONS
just changing the package versions

### ADDITIONAL INFORMATION
- [x] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
I'm not sure if you've run the application with this version of Cryptography, but we've found that it has some prohibitive performance issues. @betodealmeida do you have any suggestions here about how we can keep this versioning flexible so that people can continue to use the current version if they choose?

File: setup.py
Comment (betodealmeida) [inline setup.py:81]: Can you put up a warning in the `UPDATING.md` file about this, since there are some backwards [incompatible changes](https://cryptography.io/en/latest/changelog/) from `39.x` to `41.x`?

**Review by betodealmeida — APPROVED** [review]
Safety first!

**Comment by betodealmeida** [comment]
> I'm not sure if you've run the application with this version of Cryptography, but we've found that it has some prohibitive performance issues. @betodealmeida do you have any suggestions here about how we can keep this versioning flexible so that people can continue to use the current version if they choose?

The [big performance hit was from `36.x` to `37.x`](https://github.com/pyca/cryptography/issues/7236), reading [the changelog](https://cryptography.io/en/latest/changelog/) there seems to be no problem with going from `39.x` to `41.x`, other than they way it gets compiled.

File: setup.py
Comment (eschutho) [reply setup.py:81]: @betodealmeida I saw the Python 3.6 version which shouldn't affect Superset since we're already running on a minimum of 3.9, but do you think that the versions for `OpenSSL` and `LibreSSL` could be an issue? I'm wondering if this change would be considered a potential breaking change for Superset, and if so, if we need to think about cherrying it into 3.0 or wait until 4.0. I checked my local for example, and it seems likely that we could be running LibreSSL < 3.6

Comment (eschutho) [reply setup.py:81]: cc @michael-s-molina or @rusackas

Comment (john-bodley) [inline setup.py:81]: Should we reintroduce the lower bound (which could be increased) otherwise in theory we could be installing a version older than `39.0.1`.

**Comment by rusackas** [comment]
@suryadev99 are you able to address [this](https://github.com/apache/superset/pull/24657#discussion_r1261616542)?

**Review by rusackas — CHANGES_REQUESTED** [review]
Blocking merging for the moment based on https://github.com/apache/superset/pull/24657#discussion_r1261616542

There was also a less-blocking comment above about making an addition in UPGRADING.md that would be a nice addition.

Holler if we can help address these issues further.

**Comment by suryadev99** [comment]
@rusackas  I have made the changes, please do look into it

File: UPDATING.md
Comment (john-bodley) [inline UPDATING.md:43]:
```suggestion
```

File: UPDATING.md
Comment (john-bodley) [inline UPDATING.md:26]:
```suggestion
- [24657](https://github.com/apache/superset/pull/24657) : Bumping the cryptography package to augment the OpenSSL security vulnerability. 
```

**Review by john-bodley — COMMENTED** [review]
Thanks @suryadev99 for making the changes. There's a couple of minor formatting suggestions I recommended. Once accepted I'll go ahead and merge the PR.

**Comment by suryadev99** [comment]
Hey @john-bodley  I have incorporated the suggestions

**Comment by john-bodley** [comment]
@rusackas would you mind stamping this as I believe your blocking request has been resolved.

**Review by rusackas — APPROVED** [review]
Thank you!!!

**Review by Hanae1963 — COMMENTED** [review]
Y

**Pull request merged**
